app/helpers/terms_and_conditions_helper.rb

```diff
@@ -19,6 +23,19 @@ def terms_and_conditions_activated?
     current_order.distributor.terms_and_conditions.file?
   end
 
+  def all_terms_and_conditions_already_accepted?
+    platform_tos_already_accepted? && terms_and_conditions_already_accepted?
+  end
+
+  def platform_tos_already_accepted?
+    customer_terms_and_conditions_accepted_at = spree_current_user&.
+      customer_of(current_order.distributor)&.terms_and_conditions_accepted_at
+
+    customer_terms_and_conditions_accepted_at.present? &&
+      (customer_terms_and_conditions_accepted_at >
+       TermsOfServiceFile.updated_at)
+  end
```

Comment:
There's a bit of duplication with the other helpers here. We could extract that into a service class.